Firmware/Marlin_main.cpp

```diff
@@ -6512,6 +6513,39 @@ SERIAL_PROTOCOLPGM("\n\n");
     }
     break;
 
+    /*!
+    ### M118 - Serial print <a href="https://reprap.org/wiki/G-code#M118:_Echo_message_on_host">M118: Serial print</a>
+    #### Usage
+
+        M118 [ A1 | E1 ] [ String ]
+
+    #### Parameters
+    - `A1` - Prepend // to denote a comment or action command. Hosts like OctoPrint can interpret such commands to perform special actions. See your host’s documentation.
+    - `E1` - Prepend echo: to the message. Some hosts will display echo messages differently when preceded by echo:.
+    - `String` - Message string. If omitted, a blank line will be sent.
+    */
+    case 118: {
+        bool hasE, hasA = false;
+        char *p = strchr_pointer;
+        
+        for (uint8_t i = 2; i--;) {
+          // A1, E1, and Pn are always parsed out
+          if (!((p[0] == 'A' || p[0] == 'E') && p[1] == '1')) break;
+          switch (p[0]) {
+            case 'A': hasA = true; break;
+            case 'E': hasE = true; break;
+          }
+          p += 2;
+          while (*p == ' ') ++p;
+        }
+
+        if (hasE) SERIAL_ECHO_START;
+        if (hasA) SERIAL_ECHO("//");
+
+        SERIAL_ECHOLN(p);
```

Comment:
This call relies on the fact, that `p` ends with a `\0`. Can we always assume that?

Reply:
`get_command()` terminates all strings on either newline characters, start of comments, or when its buffer is full. That should make it safe to use `strchr_pointer` and its derivative `p` as null terminated strings.